packages/blade/src/components/Collapsible/Collapsible.tsx

```diff
@@ -48,21 +48,39 @@ const Collapsible = ({
   direction = 'bottom',
   defaultIsExpanded = false,
   isExpanded,
-  onChange,
+  onExpandChange,
   testID,
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   ...styledProps
 }: CollapsibleProps): ReactElement => {
-  const [isBodyExpanded, setIsBodyExpanded] = useState(defaultIsExpanded);
+  const [isBodyExpanded, setIsBodyExpanded] = useState(isExpanded ?? defaultIsExpanded);
+
+  // Maintain a ref to the initial value of `defaultExpanded` so changing it has no effect
+  const initialDefaultExpanded = useRef(defaultIsExpanded);
+
+  const handleExpandChange = useCallback(
+    (nextIsExpanded: boolean) => {
+      if (typeof isExpanded !== 'undefined') {
+        // controlled
+        onExpandChange?.({ isExpanded: nextIsExpanded });
+      } else {
+        // uncontrolled
+        setIsBodyExpanded(nextIsExpanded);
+        onExpandChange?.({ isExpanded: nextIsExpanded });
+      }
+    },
+    [onExpandChange, isExpanded],
+  );
```

Comment:
This ceremony is for controlled/uncontrolled state right? 
Can you check if the hook which we have works? 

```ts
const [isBodyExpanded, setIsExpanded] = useControllableState({
  value: isExpanded,
  defaultValue: defaultExpanded,
  onChange: onExpandedChange
})
```

Reply:
Spent some time trying to use this but I'm likely messing something, wasn't able to make it work (might need to spend more time). Also the code diff wasn't coming out to be significant.

Reply:
Let's take is as an refactor enhancement later.